options.go

```diff
@@ -125,8 +125,29 @@ func WithHealthz() Option {
 	return func(s *SVC) error {
 		// Register live probe handler
 		s.Router.HandleFunc("/live", func(w http.ResponseWriter, r *http.Request) {
+			var errs []error
+			for n, w := range s.workers {
+				if hw, ok := w.(Aliver); ok {
+					if err := hw.Alive(); err != nil {
+						errs = append(errs, fmt.Errorf("worker %s: %s", n, err))
+					}
+				}
+			}
+			if len(errs) == 0 {
+				w.Header().Set("Content-Type", "application/json")
+				_, _ = w.Write([]byte(`{"status": "Still Alive!"}`))
+				return
+			}
+
+			s.logger.Warn("liveliness probe failed", zap.Errors("errors", errs))
+			b, err := json.Marshal(map[string]interface{}{"errors": errs})
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			w.WriteHeader(http.StatusServiceUnavailable)
```

Comment:
This starts the write process. Meaning that the headers set on the next line will be ignored. The order of these two lines should be inverted.
See https://pkg.go.dev/net/http#ResponseWriter

Reply:
Good catch, I am not sure if a specific content type is warranted here, its mainly the status code that is relevant (most of the time), but I will just flip the two lines.

Reply:
Yeah, agreed. I just commented because the content type was there :D